modPrecompiled/src/org/aion/precompiled/PrecompiledResultCode.java

```diff
@@ -0,0 +1,96 @@
+package org.aion.precompiled;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.aion.vm.api.interfaces.ResultCode;
+
+/**
+ * An enumeration representing the execution status of a transaction.
+ */
+public enum PrecompiledResultCode implements ResultCode {
+
+    SUCCESS(0, ResultCategory.SUCCESS),
+
+    INVALID_NONCE(101, ResultCategory.REJECTED),
+
+    INVALID_NRG_LIMIT(102, ResultCategory.REJECTED),
+
+    INSUFFICIENT_BALANCE(103, ResultCategory.REJECTED),
+
+    FAILURE(1, ResultCategory.FAILED),
+
+    OUT_OF_NRG(2, ResultCategory.FAILED),
+
+    BAD_INSTRUCTION(3, ResultCategory.FAILED),
+
+    BAD_JUMP_DESTINATION(4, ResultCategory.FAILED),
+
+    STACK_OVERFLOW(5, ResultCategory.FAILED),
+
+    STACK_UNDERFLOW(6, ResultCategory.FAILED),
+
+    REVERT(7, ResultCategory.REVERT),
+
+    STATIC_MODE_ERROR(8, ResultCategory.FAILED),
+
+    ABORT(11, ResultCategory.FAILED),
+
+    VM_REJECTED(-1, ResultCategory.FATAL),
+
+    VM_INTERNAL_ERROR(-2, ResultCategory.FATAL);
+
+    private enum ResultCategory {SUCCESS, REJECTED, FAILED, FATAL, REVERT }
```

Comment:
Are these categories mutually exclusive?  If not, I don't think enum is a great way to model them

Reply:
Yes they are mutually exclusive. There is only one exception to this rule, where a failure that is caused by a `revert()` is considered both `FAILED` and `REVERT`, and that is only because revert causes a state change that is otherwise not caused by normal failures. In all other cases though, and going forward into the future, these should be mutually exclusive concepts.